src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/CCJRequestedNotificationService.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.cmc.claimstore.services.notifications;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationsProperties;
+import uk.gov.hmcts.cmc.claimstore.exceptions.NotificationException;
+import uk.gov.hmcts.cmc.claimstore.models.Claim;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIMANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIM_REFERENCE_NUMBER;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.DEFENDANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.FRONTEND_BASE_URL;
+
+@Service
+public class CCJRequestedNotificationService {
+    private final Logger logger = LoggerFactory.getLogger(CCJRequestedNotificationService.class);
+
+    private final NotificationClient notificationClient;
+    private final NotificationsProperties notificationsProperties;
+
+    @Autowired
+    public CCJRequestedNotificationService(
+        final NotificationClient notificationClient,
+        final NotificationsProperties notificationsProperties
+    ) {
+        this.notificationClient = notificationClient;
+        this.notificationsProperties = notificationsProperties;
+    }
+
+    public void notifyClaimant(final Claim claim) {
+        final Map<String, String> parameters = aggregateParams(claim);
+        sendNotificationEmail(
+            claim.getSubmitterEmail(),
+            notificationsProperties.getTemplates().getEmail().getClaimantCCJRequested(),
+            parameters,
+            NotificationReferenceBuilder.CCJRequested.referenceForClaimant(claim.getReferenceNumber())
+        );
+    }
+
+    public void notifyDefendant(final Claim claim) {
+
+        claim.getClaimData().getDefendant().getEmail()
+            .orElseThrow(() -> new NotificationException(("Unknown defendant email")));
+
+        final Map<String, String> parameters = aggregateParams(claim);
+        sendNotificationEmail(
+            claim.getClaimData().getDefendant().getEmail().get(),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS3655)

Reply:
need to fix this still